codex/sales/states/downloading.nim

```diff
@@ -75,6 +75,15 @@ method run*(
       return some State(SaleErrored(error: err, reprocessSlot: false))
 
     trace "Download complete"
+    without keyId =? reservation.key, error:
+      error "Couldn't get reservation key", id = $reservation.id, error = error.msg
+
+    if updatedReservation =? await reservations.get(keyId, Reservation):
+      if updatedReservation.size != 0:
```

Comment:
This should not fail the contract tho, if this is whats going on. We should either return that extra capacity back, or just ignore it, since the user is allegedly already paying for it.